### PR TITLE
Soroban xdr value overhaul ScVal.Equals fixes

### PR DESCRIPTION
**Pull request by paulbellamy**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Fix the ScVal.Equals method for the soroban xdr value overhaul.

### Why

Part of https://github.com/stellar/soroban-tools/issues/477
Needed for https://github.com/stellar/go/issues/4789, because this change gets the `xdr` package compiling and passing it's own tests.

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: xdr/xdr_generated.go [hunk 1]
```diff
@@ -51046,7 +51046,7 @@ func (u *ScVal) DecodeFrom(d *xdr.Decoder) (int, error) {
 	switch ScValType(u.Type) {
 	case ScValTypeScvBool:
 		u.B = new(bool)
-		nTmp, err = d.Decode((*u.B))
+		nTmp, err = d.Decode(u.B)
```
Comment (2opremio) [inline xdr/xdr_generated.go:51049]: Did you fix this in xdrgen?

File: xdr/scval_test.go [hunk 2]
```diff
@@ -23,6 +23,6 @@ func TestScValEqualsCoverage(t *testing.T) {
 
 		clonedScVal := ScVal{}
 		assert.NoError(t, gxdr.Convert(shape, &clonedScVal))
-		assert.True(t, scVal.Equals(clonedScVal))
+		assert.True(t, scVal.Equals(clonedScVal), "scVal: %#v, clonedScVal: %#v", scVal, clonedScVal)
```
Comment (2opremio) [inline xdr/scval_test.go:26]: Did this test pass?

File: xdr/scval.go [hunk 3]
```diff
@@ -119,27 +51,67 @@ func (s ScVal) Equals(o ScVal) bool {
...
+func (s ScBytes) Equals(o ScBytes) bool {
+	if len(s) != len(o) {
+		return false
+	}
+	for i := range s {
+		if s[i] != o[i] {
+			return false
+		}
+	}
+	return true
```
Comment (Shaptic) [inline xdr/scval.go:112]: there's gotta be a way to leverage [`bytes.Equal`](https://pkg.go.dev/bytes#Equal) here...

File: xdr/scval.go [hunk 4]
```diff
@@ -1,84 +1,16 @@
 package xdr
 
-import "bytes"
-
-func (s Int128Parts) Equals(o Int128Parts) bool {
-	return s.Lo == o.Lo && s.Hi == o.Hi
-}
```
Comment (Shaptic) [inline xdr/scval.go:7]: did this get absorbed e.g. by the generated xdr, or something?

File: xdr/scval_test.go (hunk 2 above)
Comment (paulbellamy) [reply xdr/scval_test.go:26]: Yes.

File: xdr/scval.go (hunk 4 above)
Comment (paulbellamy) [reply xdr/scval.go:7]: Int128Parts is a struct with two u64s (no pointers). So this does the same thing as `Int128Parts{1,1} == Int128Parts{1,1}`, and isn't needed.

**Comment by Shaptic** [comment]
Just to get the cross-reference: this partially closes #4791.

File: xdr/xdr_generated.go (hunk 1 above)
Comment (2opremio) [reply xdr/xdr_generated.go:51049]: There's another problem at:

```
func (s *DiagnosticEvent) DecodeFrom(d *xdr.Decoder) (int, error) {
	var err error
	var n, nTmp int
	nTmp, err = d.Decode(s.InSuccessfulContractCall)
```

Comment (2opremio) [reply xdr/xdr_generated.go:51049]: I think we simply need to add bool support to `xdrgen`

**Pull request merged**